### DISCUSSION TITLE
Allow to configure GraphQL `naming` to PascalCase, camelCase etc.

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#20557] [Feature Requests]
### Summary

Allow to change the format of the GraphQL schema to use `PascalCase` and `CamelCase` etc. instead of the database format.

### Motivation

Most users of GraphQL are used to have `types` as `PascalCase` and `properties` as `CamelCase`.
This is because it mostly originates from the javascript ecosystem.

But GraphQL generally does not define how these fields are formatted. This is why directus just uses the internal types of the database.

I recently read through the PR (#19867) on adding type generation to the Directus SDK and saw that this PR includes a field to configure in what way these fields should be formatted. And now thought that this feature could also be useful for the standard GraphQL endpoint.

### Detailed Design

Maybe add a url parameter to the GraphQL schema endpoint like `schemaurl?naming=pascalcase`.

### Must Have:
- Ability to change the naming through the URL

### Could Have:
- Ability to change the naming through an ENV variable?

### Drawbacks

`Maintenance`?

### Alternatives

Don't do it. :D

### Adoption Strategy

Not needed, this is opt in.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
